api/src/tests/test_context.rs

```diff
@@ -65,7 +64,7 @@ pub fn new_test_context() -> TestContext {
         db_bootstrapper::maybe_bootstrap::<DiemVM>(&db_rw, &genesis, genesis_waypoint).unwrap();
     assert!(ret);
 
-    let mempool = MockSharedMempool::new_in_runtime(&db_rw, VMValidator::new(db.clone()));
```

Comment:
If I remember correctly, there are tests depending on a real validator for validating transaction submitted in test, we need it because the test need to ensure: 
1. The transaction constructed in test by json is end with expected signed transaction 
2. Fully understand the API (submit transaction) behavior and handle error case properly. Because sometime we fail txn by mempool validation, sometime we executed txn failed.

Reply:
hmm, I don't see any tests failed by running cargo x test -p diem-api, I can add some flag for test